src/main/java/cn/nukkit/Server.java

```diff
@@ -1483,11 +1486,11 @@ private void saveOfflinePlayerDataInternal(PlayerDataSerializer serializer, Comp
 
     private void convertLegacyPlayerData() {
         File dataDirectory = new File(getDataPath(), "players/");
-        Pattern uuidPattern = Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}.dat$");
 
         File[] files = dataDirectory.listFiles(file -> {
             String name = file.getName();
-            return !uuidPattern.matcher(name).matches() && name.endsWith(".dat");
+            Matcher matcher = UUID_PATTERN.matcher(name);
+            return !matcher.matches() && name.endsWith(".dat");
```

Comment:
` && name.endsWith(".dat")` is redundant, the regex already checks for the `.dat` suffix